lua/DCS-gRPC/exporters/object.lua

```diff
@@ -25,31 +25,61 @@ GRPC.exporters.position = function(pos)
 end
 
 GRPC.exporters.unit = function(unit)
-  local vector = unit:getVelocity()
-
-  local heading = math.deg(math.atan2(vector.z, vector.x))
-  if heading < 0 then
-    heading = heading + 360
-  end
-
-  local speed = math.sqrt((vector.x)^2+(vector.z)^2)
-
   return {
     id = tonumber(unit:getID()),
     name = unit:getName(),
     callsign = unit:getCallsign(),
     coalition = unit:getCoalition() + 1, -- Increment for non zero-indexed gRPC enum
     type = unit:getTypeName(),
-    position = GRPC.exporters.position(unit:getPoint()),
+    transform = GRPC.exporters.transform(unit),
     playerName = Unit.getPlayerName(unit),
     groupName = Unit.getGroup(unit):getName(),
     numberInGroup = unit:getNumber(),
-    heading = heading,
-    speed = speed,
     category = unit:getGroup():getCategory() + 1, -- Increment for non zero-indexed gRPC enum
   }
 end
 
+GRPC.exporters.transform = function(object)
+    -- https://wiki.hoggitworld.com/view/DCS_func_getPosition
+  local position = object:getPosition()
+  local coords = GRPC.exporters.position(position.p)
+  local coordsNorth = coord.LLtoLO(coords.lat + 1, coords.lon)
+
+  local u = position.p.z
+  local v = position.p.x
+  local positionNorth = GRPC.exporters.vector(coordsNorth)
+  local projection_error = math.atan2(positionNorth.x - u, positionNorth.z - v)
+
+  local forward = position.x
+  local right = position.z
+  local up = position.y
+
+  local heading = math.atan2(forward.z, forward.x)
+  local headingDegrees = math.deg(math.atan2(forward.z, forward.x))
+  if headingDegrees < 0 then
+    headingDegrees = headingDegrees + 360
+  end
+
+  local velocity = object:getVelocity()

```

Comment:
Most of the above calculations where previously done in Rust, moving them into Lua might effect the performance.

Reply:
Compared to the other stuff that runs in pure lua environments I am willing to take the risk. We can always run it against a complicated mission and see what the server reports.